### PR TITLE
Anna Smolentzov

### PR DESCRIPTION
**Pull request by asmolentzov**

### CODE REVIEW

**Comment by asmolentzov** [comment]
@iandouglas @dionew1 
Any feedback on this would be helpful! Thanks!

File: spec/models/activerecord_obstacle_course_spec.rb [hunk 1]
```diff
@@ -146,11 +148,11 @@
     expected_result = [order_1, order_2, order_3, order_4]
 
     # ----------------------- Using Ruby -------------------------
-    orders_less_than_550 = Order.all.select { |order| order.amount < 550 }
+    # orders_less_than_550 = Order.all.select { |order| order.amount < 550 }
     # ------------------------------------------------------------
 
     # ------------------ Using ActiveRecord ----------------------
-    # Solution goes here
+    orders_less_than_550 = Order.where(amount: (0..549))
```
Comment (iandouglas) [inline spec/models/activerecord_obstacle_course_spec.rb]: Refunds would be less than 0, potentially, so we'd want to something like `Order.where("amount < ?", 550)`

File: spec/models/activerecord_obstacle_course_spec.rb [hunk 2]
```diff
@@ -300,19 +302,20 @@
...
+    #     join orders on orders.user_id=users.id
+    #     join order_items ON order_items.order_id=orders.id
+    #   where order_items.item_id=#{item_8.id}
+    #   ORDER BY users.name")
+    # users = users.map {|u| u['name']}
     # ------------------------------------------------------------
 
     # ------------------ Using ActiveRecord ----------------------
-    # Solution goes here
+    users = User.distinct.joins(orders: :order_items).where(order_items: {item_id: item_8}).pluck(:name)
```
Comment (iandouglas) [inline spec/models/activerecord_obstacle_course_spec.rb]: nice work here!

File: spec/models/activerecord_obstacle_course_spec.rb [hunk 3]
```diff
@@ -323,11 +326,11 @@
     expected_result = ['Thing 1', 'Thing 4', 'Thing 5', 'Thing 7']
 
     # ----------------------- Using Ruby -------------------------
-    names = Order.last.items.all.map(&:name)
+    # names = Order.last.items.all.map(&:name)
     # ------------------------------------------------------------
 
     # ------------------ Using ActiveRecord ----------------------
-    # Solution goes here
+    names = Order.find(Order.last.id).items.pluck(:name)
```
Comment (iandouglas) [inline spec/models/activerecord_obstacle_course_spec.rb]: This makes 3 calls to the database, where `Order.last.items.pluck(:name)` would only be two

**Pull request closed without merging**